Userland/Libraries/LibC/arch/i386/setjmp.S

```diff
@@ -4,34 +4,63 @@
  * SPDX-License-Identifier: BSD-2-Clause
  */
 
+#include <bits/sighow.h>
+
+//
+// /!\ Read setjmp.h before modifying this file!
+//
+
 .global setjmp
 setjmp:
-    mov 4(%esp), %eax
-    mov %ebx, 0(%eax)
-    mov %esi, 4(%eax)
-    mov %edi, 8(%eax)
-    mov %ebp, 12(%eax)
-    lea 4(%esp), %ecx
-    mov %ecx, 16(%eax)
-    mov (%esp), %ecx
-    mov %ecx, 20(%eax)
+    movl $0, 8(%esp)        // Set val argument to 0
```

Comment:
```
01:12⎜<RT|AO:#!/bin/mksh> SerenityOS side decided to fix sig*jmp, thats great.
     ⎜    https://github.com/SerenityOS/serenity/pull/9392/commits/535cc24e1b70c8e985d8fa3edad1ff5b5f1f67fe
01:17⎜<mirabilos:#!/bin/mksh> nice
01:20⎜<mirabilos:#!/bin/mksh> https://github.com/SerenityOS/serenity/blob/535cc24e1b70c8e985d8fa3edad1ff5b5f1f
     ⎜    67fe/Userland/Libraries/LibC/arch/i386/setjmp.S#L15
01:21⎜<mirabilos:#!/bin/mksh> congratulations, they have just overflowed the stack
01:21⎜<mirabilos:#!/bin/mksh> I mean, it’s a nice idea in theory
01:21⎜<mirabilos:#!/bin/mksh> but setjmp() is called with one argument only
01:21⎜<mirabilos:#!/bin/mksh> they try to transform setjmp(jb) to sigsetjmp(jb,0)
01:21⎜<mirabilos:#!/bin/mksh> but there’s nothing on stack reserved to store that 0
01:21⎜<mirabilos:#!/bin/mksh> better do:
01:22⎜<mirabilos:#!/bin/mksh> mov ecx,4[esp]
01:22⎜<mirabilos:#!/bin/mksh> xor eax,eax
01:22⎜<mirabilos:#!/bin/mksh> jmp 1f
01:22⎜<mirabilos:#!/bin/mksh> sigsetjmp:
01:22⎜<mirabilos:#!/bin/mksh> mov ecx,4[esp]
01:22⎜<mirabilos:#!/bin/mksh> mov eax,8[esp]
01:22⎜<mirabilos:#!/bin/mksh> 1:
```

Reply:
I guess that shows I'm not proficient at all in x86 assembly, especially in tricky situations like this... I've amended the PR with what I hope is a correct fix.

Reply:
This looks-ish better (I don’t normally do AT&T syntax), thanks.